### PR TITLE
Implement URL and trivially missing URLUtils members

### LINKED ISSUE #6322: Implement the URL object.
Blocks #6321.

* Spec: <https://url.spec.whatwg.org/#api>.
* Code: `components/script/dom/url.rs`, `components/script/dom/webidls/URL.webidl` (to be created).
* Test: `/url/url-constructor.html`, `/url/interfaces.html`.

See also http://doc.servo.org/script/dom/index.html#adding-a-new-dom-interface.

### PR DESCRIPTION
**Pull request by nox**
Fixes #6322.

Still no mutation whatsoever in URLUtils.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/6371)

### CODE REVIEW

File: components/script/dom/urlhelper.rs [hunk 1]
```diff
@@ -34,6 +21,35 @@ impl UrlHelper {
...
     }
 
+    // https://url.spec.whatwg.org/#dom-urlutils-host
+    pub fn Host(url: &Url) -> USVString {
+        USVString(match url.scheme_data {
+            SchemeData::NonRelative(..) => "".to_owned(),
+            SchemeData::Relative(ref scheme_data) => {
+                let mut host = scheme_data.host.serialize();
+                if let Some(port) = scheme_data.port {
+                    write!(host, ":{}", port).unwrap();
```
Comment (Manishearth) [inline components/script/dom/urlhelper.rs]: Any reason for choosing `write!` over `format!` here?

**Comment by Manishearth** [comment]
LGTM, one minor nit-ish.

**Comment by nox** [comment]
DOM stuff is blocked until #6150 lands.

**Comment by nox** [comment]
@bors-servo: retry

#6436 landed.

**Comment by metajack** [comment]
@nox Why did you comment out the Exposes part?

**Comment by nox** [comment]
They are commented out pretty much everywhere so I assumed they are not supported yet.

**Comment by metajack** [comment]
Uncommenting it worked for me. @Ms2ger or @jdm?

**Comment by Ms2ger** [comment]
`Exposed` is not supported in our codegen (#2823), but it is in the WebIDL parser. We should probably error on encountering it for now.

**Comment by nox** [comment]
Assigned #2823 to myself, I will try to do the correct thing there.

**Comment by metajack** [comment]
So Till and I tried uncommenting that, and it made some things work. So it does something, or at least seemed to.

**Comment by jdm** [comment]
"Some things"?

**Pull request merged**